src/shared/components/oncoprint/OncoprintUtils.ts

```diff
@@ -601,8 +715,90 @@ export function makeHeatmapTracksMobxPromise(oncoprint:ResultsViewOncoprint, sam
+                        if (!molecularProfileIdToHeatmapTracks.has(molecularProfileId)
+                            && oncoprint.sortMode.type === "heatmap"
+                            && oncoprint.sortMode.clusteredHeatmapProfile === molecularProfileId
+                        ) {
+                            oncoprint.sortByData();
+                        }
+                    })
+                };
+            });
+        },
+        default: []
+    });
+}
+
+function makeTreatmentProfileHeatmapTracksMobxPromise(oncoprint:ResultsViewOncoprint, sampleMode:boolean) {
+    return remoteData<IHeatmapTrackSpec[]>({
+        await:()=>[
+            oncoprint.props.store.samples,
+            oncoprint.props.store.patients,
+            oncoprint.props.store.molecularProfileIdToMolecularProfile,
+            oncoprint.props.store.treatmentMolecularDataCache,
+            oncoprint.props.store.treatmentLinkMap
+        ],
+        invoke:async()=>{
+
+            const molecularProfileIdToMolecularProfile = oncoprint.props.store.molecularProfileIdToMolecularProfile.result!;
+            const molecularProfileIdToHeatmapTracks = oncoprint.molecularProfileIdToHeatmapTracks;
+            const treatmentLinkMap = oncoprint.props.store.treatmentLinkMap.result!;
+
+            const treatmentProfiles = _.filter(molecularProfileIdToHeatmapTracks.values(), d => d.molecularAlterationType === AlterationTypeConstants.TREATMENT_RESPONSE);
+            const neededTreatments = _.flatten(treatmentProfiles.map(v=>v.entities.keys()));
+            await oncoprint.props.store.treatmentCache.getPromise(neededTreatments.map(g=>({treatmentId:g})), true);
+            
+            const cacheQueries = _.flatten(treatmentProfiles.map(entry=>(
+                entry.entities.keys().map(g=>({
+                    molecularProfileId: entry.molecularProfileId,
+                    treatmentId: oncoprint.props.store.treatmentCache.get({ treatmentId:g })!.data!.treatmentId,
+                    treatmentName: oncoprint.props.store.treatmentCache.get({ treatmentId:g })!.data!.name
+                }))
+            )));
+
+            await oncoprint.props.store.treatmentMolecularDataCache.result!.getPromise(cacheQueries, true);
+
+            const samples = oncoprint.props.store.samples.result!;
+            const patients = oncoprint.props.store.patients.result!;
+
+            const tracks = cacheQueries.map(query=>{
+
+                const molecularProfileId = query.molecularProfileId;
+                const profile = molecularProfileIdToMolecularProfile[molecularProfileId];
+                const dataCache = oncoprint.props.store.treatmentMolecularDataCache.result!;
+
+                const treatmentId = query.treatmentId;
+                const pivotThreshold = profile.pivotThreshold;
+                const sortOrder = profile.sortOrder;
+
+                return {
+                    key: `HEATMAPTRACK_${molecularProfileId},${treatmentId}`,
```

Comment:
would be better to prefix with `TREATMENTHEATMAPTRACK` in case theres some unlikely overlap between treatment ids and gene ids

Reply:
Yes, prefix added.